plugins/modules/repository_set_info.py

```diff
@@ -0,0 +1,166 @@
+    type: bool
+  enabled:
+    description:
+      - When true, only return repository sets which are already enabled. Defaults to false.
+    required: false
+    default: false
+    type: bool
+extends_documentation_fragment:
+  - theforeman.foreman.foreman
+  - theforeman.foreman.foreman.katelloinfomodule
+  - theforeman.foreman.foreman.infomodule
+'''
+
+EXAMPLES = '''
+- name: "Find repository set by name and product. Match repositories with the given releasever and basearch only"
+  theforeman.foreman.repository_set_info:
+    username: "admin"
+    password: "changeme"
+    server_url: "https://foreman.example.com"
+    organization: "Default Organization"
+    name: "Red Hat Enterprise Linux 7 Server (RPMs)"
+    product: "Red Hat Enterprise Linux Server"
+    repositories:
+      - releasever: "7Server"
+        basearch: "x86_64"
+      - releasever: "7.9"
+        basearch: "x86_64"
+
+- name: "Find repository set by name and product. Match all repositories in the set"
+  theforeman.foreman.repository_set_info:
+    username: "admin"
+    password: "changeme"
+    server_url: "https://foreman.example.com"
+    organization: "Default Organization"
+    name: "Red Hat Enterprise Linux 7 Server (RPMs)"
+    product: "Red Hat Enterprise Linux Server"
+    all_repositories: true
+
+- name: "Find repository set by label. Match all enabled repositories in the set"
+  theforeman.foreman.repository_set_info:
+    username: "admin"
+    password: "changeme"
+    server_url: "https://foreman.example.com"
+    organization: "Default Organization"
+    label: rhel-7-server-rpms
+    all_repositories: true
+    enabled: true
+'''
+
+RETURN = '''
+repository_sets:
+  description: List of repository sets.
+  returned: success
+  type: list
+  elements: dict
+'''
+
+
+from ansible_collections.theforeman.foreman.plugins.module_utils.foreman_helper import KatelloInfoAnsibleModule
+
+
+class KatelloRepositorySetInfo(KatelloInfoAnsibleModule):
+    pass
+
+
+def main():
+    module = KatelloRepositorySetInfo(
+        foreman_spec=dict(
+            product=dict(type='entity', scope=['organization']),
+            name=dict()
+            label=dict()
+            repositories=dict(type='list', elements='dict', options=dict(
+                basearch=dict(),
+                releasever=dict(),
+            )),
+            all_repositories=dict(type='bool'),
+            enabled=dict(type='bool'),
+        ),
+        entity_opts={'scope': ['product']},
+        required_one_of=[
+            ['label', 'name'],
+            ['repositories', 'all_repositories'],
+        ],
+        required_if=[
+            ['all_repositories', False, ['repositories']],
+            ['repositories', [], ['all_repositories']],
+        ],
+    )
+
+    # KatelloInfoAnsibleModule automatically adds organization to the entity scope
+    # but repository sets are scoped by product (and these are org scoped)
+    module.foreman_spec['entity']['scope'].remove('organization')
```

Comment:
I copied this over from the `repository_info` module but I *think* it applies here as well. I will have a better idea after writing some tests.